pkg/controller/deployment/sync_test.go

```diff
@@ -315,6 +337,16 @@ func TestScale(t *testing.T) {
 			// Get all the UPDATE actions and update nameToSize with all the updated sizes.
 			for _, action := range fake.Actions() {
 				rs := action.(testclient.UpdateAction).GetObject().(*apps.ReplicaSet)
+				if annotations, ok := test.expectedOldReplicasetAnnotations[rs.Name]; ok {
```

Comment:
It seems that with or without this patch, the added test can pass, 
Maybe take it outside of `for _, action := range fake.Actions()`